lib/fill/index.js

```diff
@@ -0,0 +1,13 @@
+import curryN from '../curryN'
+import toArray from '../toArray'
+
+export default curryN(2, function fill(val, start, end, arr) {
+  var last = arguments[arguments.length - 1]
+  return Array.isArray(last)
+    ? val
+      ? toArray(last).fill(val, Array.isArray(start) ? undefined : start, Array.isArray(end) ? undefined : end)
+      : last
+    : function(arrC) {
+        return fill(val, start, end, arrC)
+      }
```

Comment:
what does it mean? function returns function? it's not HOC.

Reply:
as @Kelin2025 said in https://github.com/Kelin2025/nanoutils/pull/14#discussion_r164082942 we expect function could be called like:
`fill(val)(arr)`
`fill(val, start)(arr)`
`fill(val, start, end)(arr)`
`fill(val, start, end, arr)`

So, if `arr` was provided through currying, we should return function to take `arr` as argument and put it in new `fill`

Maybe we do not need `curryN(2,...)` wrapper in this situation? It's size would be 142B

Reply:
Yeah, it might be reasonable

Reply:
can we use bind here? Will it be better?

Reply:
I know easy solution, I'll publish it later